From 13a369981881d2a0605ef875ce88dbf7df1c9c65 Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Mon, 22 Oct 2018 20:30:50 -0700
Subject: [PATCH 0484/1851] xilinx: v4l: scd: Validate interrupt number

commit 54fb0634211ad7ccd96a72ba50fd632d72069776 from
https://github.com/Xilinx/linux-xlnx.git

This patch ensures that the interrupt number is valid.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-scenechange-channel.c | 6 ++++++
 drivers/media/platform/xilinx/xilinx-scenechange-dma.c     | 5 +++++
 drivers/media/platform/xilinx/xilinx-scenechange.c         | 5 +++++
 3 files changed, 16 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
index 1bfd9fad6a09..6a86213d3a14 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-channel.c
@@ -351,7 +351,13 @@ static int xscd_chan_parse_of(struct xscd_chan *chan)
 	shared_data = (struct xscd_shared_data *)chan->dev->parent->driver_data;
 	shared_data->dma_chan_list[chan->id] = &chan->dmachan;
 	chan->iomem = shared_data->iomem;
+
 	chan->irq = irq_of_parse_and_map(parent_node, 0);
+	if (!chan->irq) {
+		dev_err(chan->dev, "No valid irq found\n");
+		return -EINVAL;
+	}
+
 	err = devm_request_irq(chan->dev, chan->irq, xscd_chan_irq_handler,
 			       IRQF_SHARED, dev_name(chan->dev), chan);
 	if (err) {
diff --git a/drivers/media/platform/xilinx/xilinx-scenechange-dma.c b/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
index 69edf7fcbc65..bbb588ec405d 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange-dma.c
@@ -549,7 +549,12 @@ static int xscd_dma_probe(struct platform_device *pdev)
 	xdev->common.dev = &pdev->dev;
 	ret = of_property_read_u32(node, "xlnx,numstreams",
 				   &xdev->numchannels);
+
 	irq_num = irq_of_parse_and_map(node, 0);
+	if (!irq_num) {
+		dev_err(xdev->dev, "No valid irq found\n");
+		return -EINVAL;
+	}
 
 	/* TODO: Clean up multiple interrupt handlers as there is one device */
 	ret = devm_request_irq(xdev->dev, irq_num, xscd_dma_irq_handler,
diff --git a/drivers/media/platform/xilinx/xilinx-scenechange.c b/drivers/media/platform/xilinx/xilinx-scenechange.c
index c32ea3d88a31..c07b9e6d568d 100644
--- a/drivers/media/platform/xilinx/xilinx-scenechange.c
+++ b/drivers/media/platform/xilinx/xilinx-scenechange.c
@@ -124,6 +124,11 @@ static int xscd_parse_of(struct xscd_device *xscd)
 		return ret;
 
 	xscd->irq = irq_of_parse_and_map(node, 0);
+	if (!xscd->irq) {
+		dev_err(xscd->dev, "No valid irq found\n");
+		return -EINVAL;
+	}
+
 	ret = devm_request_irq(xscd->dev, xscd->irq, xscd_irq_handler,
 			       IRQF_SHARED, dev_name(xscd->dev), xscd);
 
-- 
2.31.1

